source/attribute/Attribute.cpp

```diff
@@ -0,0 +1,456 @@
+/* Attribute.cpp
+Copyright (c) 2023 by tibetiroka
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "Attribute.h"
+
+#include <algorithm>
+#include <limits>
+#include <map>
+
+using namespace std;
+
+const string Attribute::effectNames[] = {"shields", "hull", "thrust", "reverse thrust", "turn", "active cooling",
+										"ramscoop", "cloak", "cooling", "force", "energy", "fuel", "heat", "discharge", "corrosion", "leak", "burn",
+										"ion", "scramble", "slowing", "disruption", "disabled", "minable", "piercing"};
+
+// Some category names are always the same as the corresponding effect names.
+const string Attribute::categoryNames[] = {"shield generation", "hull repair",
+										GetEffectName(THRUST), GetEffectName(REVERSE_THRUST), GetEffectName(TURN), GetEffectName(ACTIVE_COOLING),
+										GetEffectName(RAMSCOOP), GetEffectName(CLOAK), "afterburner thrust", "firing", "protection",
+										"resistance", "damage", "capacity"};
```

Comment:
I feel that this is better placed in AttributeCategory, and using a map (similar to oldtoNew below) is likely more maintainable when we add new categories.

Reply:
AttributeCategory is an enum, so if I put it there, it would be a global. I could move it to AttributeAccessor, if that would be better.